api/nodes/serializers.py

```diff
@@ -375,3 +377,57 @@ class Meta:
     @staticmethod
     def get_id(obj):
         return '{}:{}'.format(obj.node._id, obj.provider)
+
+class NodeAlternativeCitationSerializer(JSONAPISerializer):
+
+    id = IDField(source="_id", read_only=True)
+    name = ser.CharField(read_only=True)
+    text = ser.CharField(read_only=True)
+
+    class Meta:
+        type_ = 'citations'
+
+    def create(self, validated_data):
+        data = self.context['request'].data
+        node = self.context['view'].get_node()
+
+        errors = []
+        name_exists = node.alternativeCitations.find(Q('name', 'eq', data['name'])
+                                                     & Q('_id', 'ne', data['id'])).count() > 0
+        if name_exists:
+            errors.append("There is already an alternative citation named '{}'".format(data['name']))
+        matching_citations = node.alternativeCitations.find(Q('text', 'eq', data['text'])
+                                                            & Q('_id', 'ne', data['id']))
+        if matching_citations.count() > 0:
+            names = "', '".join([str(citation.name) for citation in matching_citations])
+            errors.append("Citation matches '{}'".format(names))
+        if name_exists or matching_citations.count() > 0:
+            raise ValidationError(detail=errors)
+        else:
+            citation = AlternativeCitation(text=data['text'], name=data['name'])
+            citation.save()
+            node.alternativeCitations.append(citation)
+            node.save()
+            return citation
+
+    def update(self, instance, validated_data):
+        data = self.context['request'].data
+        node = self.context['view'].get_node()
+        errors = []
```

Comment:
Can you move all of the error handling that happens in both the `create` and `update` method into a separate helper function?